commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/AesEncryptionProvider.java

```diff
@@ -33,7 +36,15 @@ public class AesEncryptionProvider implements Encryption, Decryption {
 
     private final KeyGenerator aesKeyGenerator;
 
-    public AesEncryptionProvider(final KeyGenerator aesKeyGenerator) {
+    public static AesEncryptionProvider of(final RsaEncryptionProvider rsaEncryptionProvider) {
```

Comment:
I'm not sure the whole key generator code should at all be in `RsaEncryptionProvider` as it's not used there. This means, `AesEncryptionProvider` and `RsaEncryptionProvider` could be completely independent. The `KEY_SIZE` constant should also be in `AesEncryptionProvider`. Maybe I'm missing something that justifies the current coupling between the two classes?

Reply:
Good point.
Even though we may want to keep these 2 decoupled, most users of these APIs have to initialize them together. I'm touching on that in #191 
Let me know if you feel we should decouple these here and discuss the coupling issues on #191

Reply:
Sure, let's postpone it until #191

Reply:
Oh, I just realized, RSA is not required for AES at all! Let me refactor it based on this suggestions and remove the coupling then

Reply:
Yeah, they meet each other only when we encrypt AES keys with RSA.